### PR TITLE
Coreboot risc v

### PR DESCRIPTION
**Pull request by wniewiadomski**

### CODE REVIEW

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  - Firmware
+  - coreboot
+  - RISC-V
+categories:
+  - Firmware
+  - Miscellaneous
+
+---
+
+Your post content
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:21]: ??

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+# Building coreboot for RISC-V and running it in Qemu
+
+#### 1. Install required tools and libraries
+
+On Debian:
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:27]: Whic OS have you tried? Write it down. Including Arch and Fedora does not make much sense if you have not tested it. For example, the qemu packages are missing from arch/fedora as far as I can see.

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+```sh
+$ sudo dnf install git make gcc-gnat flex bison xz bzip2 gcc g++ ncurses-devel wget zlib-devel
+```
+
+#### 2. Download coreboot source tree
+
+```sh
+$ git clone https://review.coreboot.org/coreboot
+$ cd coreboot
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:48]: Leave a note which revision was used at the time of writing. Default branch (master?) is a moving target. In a few months something might work differently

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+At the end of the process, you should get the info that your cross toolchain is
+built
+
+```
+You can now run riscv64-elf cross GCC from /your/path/to/coreboot/util/crossgcc/xgcc.
+```
+
+#### 4. Configure payload
+
+> Unfortunately using demonstration payloads such as `coreinfo` or `tint` is not possible as they use `libpayload` library which does not support RISC-V architecture yet. However, there is a [WIP branch](https://review.coreboot.org/c/coreboot/+/31356) working on adding initial support for RISC-V you can check out.
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:70]: wram at 80 chars

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+
+As an example payload I will use SeaBIOS, which you can later use to boot the operating system.
+
+
+Firstly you need to clone the repository.
+
+```sh
+$ git clone https://review.coreboot.org/seabios.git
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:79]: provide the git revision which was used

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+```
+
+You should get the output file in `out/bios.bin.elf`
+
+#### 5. Configure the build
+
+Configure your mainboard
+
+```sh
+$ make menuconfig
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:110]: AIs this in seabios or coreboot repo? It is not clear from this document.

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+The output should look like this:
+
+```
+CONFIG_BOARD_EMULATION_QEMU_RISCV_RV64=y
+CONFIG_PAYLOAD_ELF=y
+CONFIG_PAYLOAD_FILE="seabios/out/bios.bin.elf"
+```
+
+#### 6. Build coreboot
+
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:144]: avoid empty lines (unless they are specifically needed in this case for some reason?)

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+CBFS: Locating 'fallback/payload'
+CBFS: Found @ offset 9740 size 11187
+Checking segment from ROM address 0x80029978
+SELF segment doesn't target RAM: 0x000dfa80, 132480 bytes
+ 0. 0000000080400000-0000000080431fff: RAM
+ 1. 0000000080432000-0000000080446fff: RAMSTAGE
+ 2. 0000000080447000-0000000081431fff: RAM
+ 3. 0000000081432000-0000000081432fff: RAMSTAGE
+ 4. 0000000081433000-00000000bffdbfff: RAM
+ 5. 00000000bffdc000-00000000bfffffff: CONFIGURATION TABLES
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:256]: Is this only coreboot log or seabios as well?

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+```
+
+
+## Summary
+
```
Comment (macpijan) [inline blog/content/post/2020-08-26-coreboot-riscv.md:261]: Some short summary from you on the post topic would be nice.

**Review by macpijan — CHANGES_REQUESTED** [review]

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+  - Firmware
+  - Miscellaneous
+
+---
+
+Your post content
+
+# Building coreboot for RISC-V and running it in Qemu
+
+#### 1. Install required tools and libraries
```
Comment (miczyg1) [inline blog/content/post/2020-08-26-coreboot-riscv.md:25]: This should be a blog post, NOT a guide/procedure how to build. Some introduction is highly recommended that will briefly explain what this blog post will be about, what is RISC-V and why is this architecture so exciting.

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+$ cd coreboot
+```
+
+#### 3. Build coreboot toolchain
+
+You can use CPUS= to specify the number of threads you want to use for building
+toolchain.
+
+```sh
+$ make crossgcc-riscv CPUS=4
```
Comment (miczyg1) [inline blog/content/post/2020-08-26-coreboot-riscv.md:57]: This is often errorprone. Use docker instead:

```
docker pull coreboot/coreboot-sdk:65718760fa
docker run --rm -it -v $PWD:/home/coreboot/coreboot -w /home/coreboot/coreboot coreboot/coreboot-sdk:65718760fa /bin/bash
```

And your environment is ready

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+> NOTE: This will take a while
+
+At the end of the process, you should get the info that your cross toolchain is
+built
+
+```
+You can now run riscv64-elf cross GCC from /your/path/to/coreboot/util/crossgcc/xgcc.
+```
+
+#### 4. Configure payload
```
Comment (miczyg1) [inline blog/content/post/2020-08-26-coreboot-riscv.md:68]: This step is obsolete. coreboot integrates payloads automatically. And I'm not sure SeaBIOS is supported on RISC-V

File: blog/content/post/2020-08-26-coreboot-riscv.md (hunk 8 above)
Comment (miczyg1) [reply blog/content/post/2020-08-26-coreboot-riscv.md:256]: As said in previous comment, SeaBIOS is likely not supported on RISC-V. Try Linux payload or LinuxBoot

File: blog/content/post/2020-08-26-coreboot-riscv.md (hunk 6 above)
Comment (miczyg1) [reply blog/content/post/2020-08-26-coreboot-riscv.md:110]: One should not clone any payload themselves, coreboot already integrates building payload on its own. Secondly SeaBIOS will not work with RISC-V

File: blog/content/post/2020-08-26-coreboot-riscv.md [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+
+Inside `menuconfig` follow these steps:
+
+```
+   select 'Mainboard' menu
+   select '(Emulation)' in 'Mainboard vendor'
+   select 'QEMU RISC-V rv64' in 'Mainboard model'
+   select < Exit >
+   select `Payload` menu
+   select '(An ELF executable payload)' in 'Add a payload'
```
Comment (miczyg1) [inline blog/content/post/2020-08-26-coreboot-riscv.md:121]: Choose LinuxBoot or Linux payload. Although it will require small configuration

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Review by macpijan — APPROVED** [review]

File: blog/content/post/2020-09-28-coreboot-riscv.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+getting so successful is while x86 or ARM require a license to be used, RISC-V
+can be implemented by anyone for free and companies can modify it to fit their
+needs, which makes them independent from the main providers and may lead to
+increase of competitiveness in the aspect of innovation.  
+While getting more and more attention, RISC-V is also getting more support. It
+is supported architecture for coreboot. In the next steps, I will explain how to
+build coreboot for RISC-V and run it in Qemu emulator.
+
+
+#### 2. Build Docker image
```
Comment (miczyg1) [inline blog/content/post/2020-09-28-coreboot-riscv.md:41]: D
```suggestion
#### 2. Download and run Docker image
```

**Review by miczyg1 — APPROVED** [review]

**Review by artur-rs — APPROVED** [review]

**Pull request merged**